### ISSUE TITLE
Since 3.6.0 AutoEncoder does not take quality as a constructor parameter

### ISSUE THREAD

**Issue by karlomikus** [Intervention/image#1332]
**Describe the bug**
Not sure if this is breaking change, but if you follow docs, the examples are not correct anymore: https://image.intervention.io/v3/basics/image-output#encoding-images

**Code Example**
The following code is not working anymore:
```php
$encodedImage = $image->encode(new AutoEncoder(quality: 85));
```

**Environment (please complete the following information):**
 - PHP Version: 8.2.0
 - OS: Ubuntu
 - Intervention Image Version: 3.6.0
 - GD or Imagick: Imagick

**Comment by olivervogel** [comment]
Thanks for reporting. Unfortunately I can't reproduce this at the moment. What is the error message you are getting?

**Comment by karlomikus** [comment]
This happens when I try to encode `.png` image.

```log
[2024-04-22 14:11:41] local.ERROR: [IMAGE_SERVICE] Unknown named parameter $quality  
```

I guess this happens because `PngEncoder` does not have `$quality` parameter.

**Comment by olivervogel** [comment]
Ok, I see your point. This needs to be adressed, I will look into it.

**Comment by olivervogel** [comment]
This should be fixed in `3.6.1`.
